frontend/packages/knative-plugin/src/components/add/event-sources/EventSourcesSelector.tsx

```diff
@@ -4,23 +4,28 @@ import { useFormikContext, FormikValues } from 'formik';
 import { ItemSelectorField } from '@console/shared';
 import FormSection from '@console/dev-console/src/components/import/section/FormSection';
 import { NormalizedEventSources } from '../import-types';
-import { getEventSourceData } from '../../../utils/create-eventsources-utils';
+import {
+  isKnownEventSourceSelected,
+  getEventSourceData,
+} from '../../../utils/create-eventsources-utils';
 
 interface EventSourcesSelectorProps {
   eventSourceList: NormalizedEventSources;
 }
 
 const EventSourcesSelector: React.FC<EventSourcesSelectorProps> = ({ eventSourceList }) => {
-  const { setFieldValue, setFieldTouched, validateForm } = useFormikContext<FormikValues>();
   const eventSourceItems = Object.keys(eventSourceList).length;
+  const { setFieldValue, setFieldTouched, validateForm } = useFormikContext<FormikValues>();
   const handleItemChange = React.useCallback(
     (item: string) => {
-      const nameData = `data.${item.toLowerCase()}`;
-      const sourceData = getEventSourceData(item.toLowerCase());
-      setFieldValue(nameData, sourceData);
-      setFieldTouched(nameData, true);
-      setFieldValue('name', _.kebabCase(item));
-      setFieldTouched('name', true);
+      if (isKnownEventSourceSelected(item)) {
```

Comment:
do we need this `if` block.. ? and what's the use of just `validateForm();` outside of if block

Reply:
I just intended to skip that part unless it is a form
moved `validateForm()` and the code to set the name out of the if block